auto_submit/lib/requests/check_pull_request.dart

```diff
@@ -27,23 +27,56 @@ class CheckPullRequest extends RequestHandler {
     final String rawBody = await request.readAsString();
     final body = json.decode(rawBody) as Map<String, dynamic>;
     final PullRequest pullRequest = PullRequest.fromJson(body['pull_request']);
+    final RepositorySlug slug = pullRequest.base!.repo!.slug();
 
     final GithubService gitHub = await config.createGithubService();
     final _AutoMergeQueryResult queryResult = await _parseQueryData(pullRequest, gitHub);
-    if (await shouldMergePullRequest(queryResult)) {
+    if (await shouldMergePullRequest(queryResult, slug, gitHub)) {
       // TODO(Kristin): Keep pulling pubsub queue, https://github.com/flutter/flutter/issues/98704
 
-    } else {
+    } else if (queryResult.shouldRemoveLabel) {
+      log.info('Removing label for commit: ${queryResult.sha}');
+      await _removeLabel();
       return Response.ok(jsonEncode(<String, String>{}));
     }
 
     return Response.ok(rawBody);
   }
 
   /// Check if the pull request should be merged.
-  Future<bool> shouldMergePullRequest(_AutoMergeQueryResult queryResult) async {
-    // TODO(Kristin): Implement shouldMergePullRequest, https://github.com/flutter/flutter/issues/98707
+  ///
+  /// A pull request should be merged on either cases:
+  /// 1) All tests have finished running and satified basic merge requests
```

Comment:
I don't think (1) has been added in this PR

Reply:
Hi Casey, I think queryResult.shouldMerge implements 1). It checks the checkRuns and statuses and also some other basic merge requests like whether CI has run successfully and whether the PR has conflicts or not.
The checkRuns part is still a TODO.

Reply:
Thanks, I didn't see it in the diff (but it was added in a previous PR)